### ISSUE TITLE
cv doesn't plot more than 3 folds with custom indices

### ISSUE THREAD

**Issue by pkhokhlov** [catboost/catboost#1961]
Problem: cv does not plot more than 3 folds when custom splitting indices given
catboost version: 1.0.3
Operating System: ubuntu
CPU: Intel

Reproducible example:
```
import catboost
from sklearn.datasets import make_classification
import numpy as np

np.random.seed(1)

X, y = make_classification(random_state=1)

# can change this
n_folds = 10
assert(X.shape[0] % n_folds == 0)

fold_assignment = np.repeat(range(n_folds), X.shape[0] / n_folds)
np.random.shuffle(fold_assignment)

fold_idxs = []
for i in range(n_folds):
    fold_idxs.append((
        np.argwhere(fold_assignment == i).flatten().tolist(),
        np.argwhere(fold_assignment != i).flatten().tolist()
    ))

params = {
    'n_estimators': 10,
    'loss_function': 'Logloss',
}
pool = catboost.Pool(X, y)
print(f'we expect {len(fold_idxs)} folds in the plot')

models, scores = catboost.cv(pool,
                             params,
                             folds=iter(fold_idxs),
                             logging_level='Silent',
                             plot=True,
                             return_models=True,
                            )
print(f'number of models returned: {len(models)}')
```
If `n_folds` is changed to 2 or 3, then the plot correctly shows only 2 or 3 folds. If there are more than 3 folds, the plot does not show folds > 3, but the returned models/scores suggest those folds are evaluated.

**Comment by LyzhinIvan** [comment]
Hi, @pkhokhlov!

First of all, you mixed up the return values in places - `catboost.cv` returns `(scores, models)`, not `(models, scores)`.

Secondly, thanks fo reporting! We will fix it soon. Right now you can plot learning curves after cross-validation with code like this:
```
from catboost import MetricVisualizer

vizualizer = MetricVisualizer(train_dirs=[f'catboost_info/fold-{i}' for i in range(n_folds)])
vizualizer.start()
```

**Comment by LyzhinIvan** [comment]
Fixed in 7bf09c449d146b32a010f950ab26c28d0dd17b72.

**Comment by Tesax123** [comment]
@LyzhinIvan 
I have the same issue, but with a generator object (GroupKFold from scikit learn) instead of custom indices. 
I'm on version 1.0.4 of catboost. 

I don't have a reproducible example, sorry, but this is the part of my code that shows a plot with only 3 folds plotted instead of the expected 5.



```
cv_dataset = Pool(data=X, label=y)

params = {
    'iterations': 300,
    'learning_rate': 0.1,
    'random_seed': 42,
    'eval_metric': metrics.Accuracy(),
    'logging_level': 'Silent',
    'depth': 6,
    'class_weights': class_weight,
    'loss_function':'MultiClass'
}

scores = cv(cv_dataset,
            params,
            folds= GroupKFold(n_splits=5).split(X=X, y=y, groups=groups),
            plot="True")
```

Any help would be greatly appreciated. 

![image](https://user-images.githubusercontent.com/48088982/158028946-70e5fc0f-4faa-4b93-9158-9b75b711a206.png)

**Comment by LyzhinIvan** [comment]
Hi, @Tesax123! My fix was merged after the last release, so it will be available since 1.0.5 only. I think it should happen soon. Right now you can use MetricVisualizer as in the example above.
